ts/types/MessageWithContentPO.ts

```diff
@@ -0,0 +1,46 @@
+import * as t from "io-ts";
+
+import { MessageBodyMarkdown } from "../../definitions/backend/MessageBodyMarkdown";
+import { MessageSubject } from "../../definitions/backend/MessageSubject";
+import { MessageWithContent } from "../../definitions/backend/MessageWithContent";
+
+// Required attributes
+const MessageWithContentPOR = t.interface({
+  created_at: t.string,
+
+  id: t.string,
+
+  sender_service_id: t.string
+});
+
+// Optional attributes
+const MessageWithContentPOO = t.partial({
+  markdown: MessageBodyMarkdown,
+
+  subject: MessageSubject
+});
+
+/**
+ * A plain object representation of a MessageWithContent useful to avoid problems with the redux store.
+ * The create_at date object is transformed in a string.
+ */
+export const MessageWithContentPO = t.exact(
+  t.intersection(
+    [MessageWithContentPOR, MessageWithContentPOO],
+    "MessageWithContentPO"
+  )
+);
+
+export type MessageWithContentPO = t.TypeOf<typeof MessageWithContentPO>;
```

Comment:
can this type be derived from `MessageWithContent` by overriding just the `created_at` type? I don't like the duplication

Reply:
@gunzip

Reply:
yes web can using .props, ie:

```ts
import * as t from "io-ts";

const orig = t.type({
  created_at: t.number,
  dummy: t.string
});

const override = t.type({ ...orig.props, created_at: t.string });
```